src/System.Diagnostics.Process/tests/ProcessStartInfoTests.cs

```diff
@@ -496,8 +541,11 @@ public void TestEnvironmentVariables_Environment_DataRoundTrips()
             psi.EnvironmentVariables.Add("NewKey3", "NewValue3");
             KeyValuePair<string, string>[] kvpa = new KeyValuePair<string, string>[5];
             psi.Environment.CopyTo(kvpa, 0);
-            Assert.Equal("NewKey3", kvpa[2].Key);
-            Assert.Equal("NewValue3", kvpa[2].Value);
+
+            // .NET Core uses a Dictionary<string, string>, but the full .NET Framework uses
+            // StringDictionary. These collections order the output of CopyTo differently.
+            Assert.Equal(PlatformDetection.IsFullFramework ? "NewKey2" : "NewKey3", kvpa[2].Key);
```

Comment:
since the order of hte dictionary is not relevant in the first place (or even well defined) why not just throw in an OrderBy and ther'es no need to check what FX you're running on?